spec/functional/resource/powershell_spec.rb

```diff
@@ -99,6 +122,18 @@
     it "returns 1 if the last command was a cmdlet that failed and was preceded by a successfully executed non-cmdlet Windows binary" do
       resource.code([windows_process_exit_code_success_content, cmdlet_exit_code_not_found_content].join(';'))
       resource.returns(1)
+      expect { resource.run_action(:run) }.not_to raise_error
+    end
+
+    it "raises an error if the script is not syntactically correct and returns is not set to 1" do
+      resource.code('if({)')
+      resource.returns(0)
+      expect { resource.run_action(:run) }.to raise_error(Mixlib::ShellOut::ShellCommandFailed)
+    end
+
+    it "returns 1 if the script provided to the code attribute is not syntactically correct" do
+      resource.code('if({)')
+      resource.returns(1)
       resource.run_action(:run)
```

Comment:
There's no expect in here. I messed with this and ran it, and I realized/remembered that Mixlib raises an error when the return code is different than what is specified. So if you change the "returns" line, this test fails because an error is raised.

All the same, I'm thinking it would be more clear if this line was:
```
expect { resource.run_action(:run) }.not_to raise_error
```